server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregator.java

```diff
@@ -279,8 +280,8 @@ private void collectZeroDocEntriesIfNeeded(long owningBucketOrd) throws IOExcept
             MultiTermsValuesSourceCollector collector = multiTermsValue.getValues(ctx);
             // brute force
             for (int docId = 0; docId < ctx.reader().maxDoc(); ++docId) {
-                for (List<Object> value : collector.apply(docId)) {
-                    bucketOrds.add(owningBucketOrd, encode(value));
+                for (BytesRef compositeKey : collector.apply(docId)) {
+                    bucketOrds.add(owningBucketOrd, compositeKey);
```

Comment:
I suspect the `MultiTermsAggregator::encode` could be dropped, but shouldn't the `decode` part also be changed as well since we now store ByteRef and not List<Object>?

Reply:
Ah, yes! `encode` should be removed.

The output of encode was a BytesRef, i.e. the encoded composite key. Now we have moved the encoding logic within the `MultiTermsValuesSourceCollector` itself and the output is binary-compatible with the previous format. So the same `decode` method can be reused to parse it back into a list of arbitrary Java objects.